site/js/ledAPI.js

```diff
@@ -0,0 +1,256 @@
+                    document.getElementById(lastDevice).classList.remove(
+                        'btn-current');
+                }
+            }
+
+            console.debug("setting device button to current");
+            document.getElementById(deviceID).classList.add(
+                'btn-current');
+        });
+
+
+        /* Change name of device when form is submitted */
+        $('form').on('submit', function changeDeviceName(event) {
+
+            /* Prevent the page from reloading after form submit */
+            event.preventDefault();
+
+            /* Set the text-output span to the value of the first input */
+            var $input = $(this).find('input');
+            var input = $input.val();
+
+            /* update device name on server */
+            $apiUtils.putRequest(apiUrl, deviceID, 'name', input);
+
+            /*  update displayed name for device button */
+            $('#currentDeviceLabel').html(input.slice(0,
+                nameMaxLength));
+            document.getElementById(deviceID).value = input.slice(0,
+                nameMaxLength);
+
+        });
+
+    }
+
+    function updateOnButton(device) {
+        console.debug("updateOnButton: state received: ", device[1][
+            'onState'
+        ]);
+        switch (device[1]['onState']) {
+            case "true":
+                $("#onSwitch").prop("checked", true);
+                break;
+            case "false":
+                $("#onSwitch").prop("checked", false);
+                break;
+        }
+    }
+
+
+    function updateBrightSlider(device) {
+        var brightVal = device[1]['brightness']
+        $('#brightness-value').html(brightVal);
+        $('#slider-brightness').val(brightVal);
+    }
+
+    function updateDeviceList(newDevices) {
+        var remove = [];
+        var add = [];
+
+        newDevices.sort();
+
+        /* Identify old devices no longer online and remove */
+        for (var i in devices) {
+            if (newDevices.indexOf(devices[i]) === -1) remove.push(devices[
+                i]);
+        }
+        remove.forEach(removeDevice);
+
+        /* Hide parameter UI and current device label if not online (e.g., no heartbeat) */
+        if (newDevices.includes(("hb_" + deviceID)) == false) {
+            $("#currentDeviceLabel").hide();
+            $("#parameter-UI").toggleClass('look-disabled', true);
+            console.debug("hiding currentDeviceLabel");
+        } else {
+            console.debug(
+                "getting/showing currentDeviceLabel and parameters");
+
+            /* if current device is in heartbeat list, get device info */
+            $apiUtils.getData(apiUrl, deviceID, updateMainContent,
+                disableMainContent);
+
+            $("#currentDeviceLabel").show();
+            $("#parameter-UI").toggleClass('look-disabled', false);
+        }
+
+        /* Identify new devices received in order to add */
+        for (var i in newDevices) {
```

Comment:
Same assumptions as previous comment
```suggestion
        const deviceSet = new Set(devices);
        for (const device of newDevices)
            if (!deviceSet.has(device)) 
                add.push(device);
```